pkg/sql/logictest/tests/local-mixed-22.2-23.1/local-mixed-22_2-23_1_test: TestLogic_synthetic_privileges failed

pkg/sql/logictest/tests/local-mixed-22.2-23.1/local-mixed-22_2-23_1_test.TestLogic_synthetic_privileges [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_SqlLogicTestHighVModuleNightlyBazel/11339546?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_SqlLogicTestHighVModuleNightlyBazel/11339546?buildTab=artifacts#/) on master @ [e0de1f12d4f496fc7e4050a33e96f8c635c7a27c](https://github.com/cockroachdb/cockroach/commits/e0de1f12d4f496fc7e4050a33e96f8c635c7a27c):


```
--- End jemalloc statistics ---
[07:11:09] CREATE USER testuser;
[07:11:09] GRANT SYSTEM MODIFYCLUSTERSETTING TO testuser;
[07:11:10] GRANT SELECT ON crdb_internal.tables TO testuser;
[07:11:10] GRANT USAGE ON EXTERNAL CONNECTION foo TO testuser;
[07:11:10] DROP USER testuser;
[07:11:10] GRANT SYSTEM MODIFYCLUSTERSETTING TO testuser2;
[07:11:10] --- progress: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/4110/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/sql/logictest/tests/local-mixed-22.2-23.1/local-mixed-22_2-23_1_test_/local-mixed-22_2-23_1_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/synthetic_privileges: 57 statements
[07:11:10] GRANT USAGE ON EXTERNAL CONNECTION foo TO testuser2;
[07:11:11] DROP USER testuser, testuser2;
[07:11:11] CREATE USER testuser3;
[07:11:11] GRANT SYSTEM MODIFYCLUSTERSETTING, EXTERNALCONNECTION TO testuser3;
[07:11:11] ALTER DEFAULT PRIVILEGES GRANT SELECT ON TABLES TO testuser3;
[07:11:11] DROP USER testuser3;
[07:11:12] CREATE USER testuser4;
[07:11:12] REVOKE SELECT ON crdb_internal.tables FROM public;
[07:11:12] SELECT has_table_privilege('testuser4', 'crdb_internal.tables', 'SELECT');
[07:11:12] 	-- OK;
[07:11:12] BEGIN;
[07:11:12] GRANT SELECT ON crdb_internal.tables TO testuser4;
[07:11:12] SELECT username, path, privileges, grant_options FROM system.privileges ORDER BY 1,2;
[07:11:12] 	-- OK;
[07:11:12] SELECT has_table_privilege('testuser4', 'crdb_internal.tables', 'SELECT');
[07:11:12] 	-- OK;
[07:11:12] ROLLBACK;
[07:11:12] SELECT username, path, privileges, grant_options FROM system.privileges ORDER BY 1,2;
[07:11:12] 	-- OK;
[07:11:12] SELECT has_table_privilege('testuser4', 'crdb_internal.tables', 'SELECT');
[07:11:12] 	-- OK;
=== CONT  TestLogic_synthetic_privileges
    logic.go:4317: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/4110/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/sql/logictest/tests/local-mixed-22.2-23.1/local-mixed-22_2-23_1_test_/local-mixed-22_2-23_1_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/synthetic_privileges:392: error while processing
    logic.go:4318: failed to close connection to node 0 for user testuser: write tcp 127.0.0.1:44634->127.0.0.1:34211: write: broken pipe
    panic.go:522: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/e8057174548ad26f2adbc87bbecdbe57/logTestLogic_synthetic_privileges439885368
--- FAIL: TestLogic_synthetic_privileges (15.48s)
=== RUN   TestLogic_synthetic_privileges/all_does_not_include_nosqllogin
[07:11:13] GRANT SYSTEM ALL TO testuser;
[07:11:14] CANCEL SESSION (SELECT session_id FROM [SHOW SESSIONS] WHERE user_name = 'testuser');
[07:11:14] SELECT 1;
[07:11:14] 	-- OK;
[07:11:14] 	-- FAIL
    logic.go:2174: 
         
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/4110/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/sql/logictest/tests/local-mixed-22.2-23.1/local-mixed-22_2-23_1_test_/local-mixed-22_2-23_1_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/synthetic_privileges:392: 
        expected success, but found
        read tcp 127.0.0.1:44634->127.0.0.1:34211: read: connection reset by peer
[07:11:14] --- done: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/4110/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/sql/logictest/tests/local-mixed-22.2-23.1/local-mixed-22_2-23_1_test_/local-mixed-22_2-23_1_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/synthetic_privileges with config local-mixed-22.2-23.1: 80 tests, 1 failures

    --- FAIL: TestLogic_synthetic_privileges/all_does_not_include_nosqllogin (0.32s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/sql-queries
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLogic_synthetic_privileges.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-30656

--- a/pkg/sql/logictest/logic.go
+++ b/pkg/sql/logictest/logic.go
@@ -3804,10 +3804,10 @@ func (t *logicTest) validateAfterTestCompletion() error {
 		if user == username.RootUser {
 			continue
 		}
-		for i, c := range userClients {
-			if err := c.Close(); err != nil {
-				t.Fatalf("failed to close connection to node %d for user %s: %v", i, user, err)
-			}
+		for _, c := range userClients {
+			// Ignore the error from closing the connection. This may not succeed if,
+			// for example, CANCEL SESSION was called on one of the sessions.
+			_ = c.Close()
 		}
 		delete(t.clients, user)
 	}

